app/controllers/api/v2/hosts_controller.rb

```diff
@@ -203,6 +203,21 @@ def facts
         render_message(e.to_s, :status => :unprocessable_entity)
       end
 
+      api :POST, "/hosts/:id/rebuild_config", N_("Rebuild TFTP, DNS DHCP configuration")
+      param :id, :identifier_dottable, :required => true
+      def rebuild_config
+        result = @host.rebuild_config
+        failures = []
+        result.each_pair do |key, value|
+          failures << key.to_s.upcase unless value
+        end
+        if failures.empty?
+          render :json => {:message => _("Configuration successfully rebuilt.")}, :status => :ok
+        else
+          render :json => {:message => _("Configuration rebuild failed for: %s.") % (failures.join(", "))}, :status => :unprocessable_entity
```

Comment:
We are not passing full exceptions when there is a fail, we only provide a client with a message what failed like [this](http://i.imgur.com/gfYzsHG.png). I did not think we had to inform there was a Net::Error, but it certainly can be done. Would you like me to?